### PR TITLE
Add support for llvm 15

### PR DESCRIPTION
**Pull request by chenhengqi**
##### Checklist

- [x] Language changes are updated in `man/adoc/bpftrace.adoc` and if needed in `docs/reference_guide.md`
- [x] User-visible and non-trivial changes updated in `CHANGELOG.md`
- [x] The new behaviour is covered by tests

### CODE REVIEW

File: src/ast/irbuilderbpf.cpp [hunk 1]
```diff
@@ -282,7 +282,11 @@ CallInst *IRBuilderBPF::createCall(Value *callee,
 {
 #if LLVM_VERSION_MAJOR >= 11
   auto *calleePtrType = cast<PointerType>(callee->getType());
+#if LLVM_VERSION_MAJOR >= 15
+  auto *calleeType = cast<FunctionType>(calleePtrType->getPointerElementType());
```
Comment (viktormalik) [inline src/ast/irbuilderbpf.cpp:286]: `Type::getPointerElementType` has been in LLVM for a long time, so we should be able to use it for all versions. In fact, here you don't even need to do the above cast to `PointerType`.

This also applies to the rest of changes in this PR.

File: src/ast/irbuilderbpf.cpp [hunk 2]
```diff
@@ -382,8 +386,13 @@ Value *IRBuilderBPF::CreateMapLookupElem(Value *ctx,
     CREATE_MEMCPY(value, call, type.GetSize(), 1);
   else
   {
+#if LLVM_VERSION_MAJOR >= 15
+    assert(value->getType()->isPointerTy() &&
+           (value->getType()->getPointerElementType() == getInt64Ty()));
```
Comment (viktormalik) [inline src/ast/irbuilderbpf.cpp:391]: Same as above: use `getPointerElementType` for all versions.

**Review by viktormalik — CHANGES_REQUESTED** [review]

**Comment by viktormalik** [comment]
`clang-format` reports some issues, please fix them.

**Comment by chenhengqi** [comment]
Done.

**Review by viktormalik — APPROVED** [review]

**Pull request merged**